ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -17062,98 +17064,99 @@ ABoundActivity * HqlCppTranslator::doBuildActivityTempTable(BuildCtx & ctx, IHql
 
     buildInstancePrefix(instance);
 
-    BuildCtx funcctx(instance->startctx);
-    funcctx.addQuotedCompoundLiteral("virtual size32_t getRow(ARowBuilder & crSelf, __uint64 row)");
-    ensureRowAllocated(funcctx, "crSelf");
-    BoundRow * selfCursor = bindSelf(funcctx, instance->dataset, "crSelf");
-    IHqlExpression * self = selfCursor->querySelector();
-    OwnedHqlExpr clearAction;
-
     OwnedHqlExpr rowsExpr;
-    OwnedHqlExpr rowVar = createVariable("row", makeIntType(8, false));
-    if (expr->getOperator() == no_datasetfromrow)
     {
-        BuildCtx subctx(funcctx);
-        subctx.addFilter(rowVar);
-        if (clearAction)
-            subctx.addExpr(clearAction);
-        subctx.addReturn(queryZero());
+        MemberFunction func(*this, instance->startctx, "virtual size32_t getRow(ARowBuilder & crSelf, __uint64 row)");
+        ensureRowAllocated(func.ctx, "crSelf");
+        BoundRow * selfCursor = bindSelf(func.ctx, instance->dataset, "crSelf");
+        IHqlExpression * self = selfCursor->querySelector();
+        OwnedHqlExpr clearAction;
 
-        buildAssign(funcctx, self, values);
-        buildReturnRecordSize(funcctx, selfCursor);
-        rowsExpr.setown(getSizetConstant(1));
-    }
-    else if ((values->getOperator() == no_list) || (values->getOperator() == no_null))
-//  else if (((values->getOperator() == no_list) && 
-//                  (!values->isConstant() || (values->queryType()->queryChildType()->getSize() == UNKNOWN_LENGTH))) || (values->getOperator() == no_null))
-    {
-        //
-        unsigned maxRows = values->numChildren();
-        if (maxRows)
+        OwnedHqlExpr rowVar = createVariable("row", makeIntType(8, false));
+        if (expr->getOperator() == no_datasetfromrow)
         {
-            unsigned dummyIdx = 0;
-            OwnedHqlExpr tgt = createSelectExpr(LINK(self), LINK(queryNextRecordField(record, dummyIdx)));
-
-            BuildCtx switchctx(funcctx);
-            switchctx.addQuotedCompoundLiteral("switch (row)");
+            BuildCtx subctx(func.ctx);
+            subctx.addFilter(rowVar);
+            if (clearAction)
+                subctx.addExpr(clearAction);
+            subctx.addReturn(queryZero());
 
-            unsigned row;
-            for (row = 0; row < maxRows; row++)
+            buildAssign(func.ctx, self, values);
+            buildReturnRecordSize(func.ctx, selfCursor);
+            rowsExpr.setown(getSizetConstant(1));
+        }
+        else if ((values->getOperator() == no_list) || (values->getOperator() == no_null))
+    //  else if (((values->getOperator() == no_list) &&
+    //                  (!values->isConstant() || (values->queryType()->queryChildType()->getSize() == UNKNOWN_LENGTH))) || (values->getOperator() == no_null))
+        {
+            //
```

Comment:
No reason, but I wouldn't normally clean up deleting lines as part of an unrelated PR.